Compression/Compression.cs

```diff
@@ -874,5 +875,44 @@ public static IEnumerable<string> GetZipEntryFileNames(string zipFileName)
                 }
             }
         }
+
+        /// <summary>
+        /// Extracts a 7-zip archive to disk, using the 7-zip CLI utility
+        /// </summary>
+        /// <param name="inputFile">Path to the 7z file</param>
+        /// <param name="outputDirectory">Directory to output contents of 7z</param>
+        /// <param name="execTimeout">Timeout in seconds for how long we should wait for the extraction to complete</param>
+        /// <exception cref="Exception">The extraction failed because of a timeout or the exit code was not 0</exception>
+        public static void Extract7ZipArchive(string inputFile, string outputDirectory = null, int execTimeout = 60)
+        {
+            outputDirectory = outputDirectory ?? $"{Guid.NewGuid()}";
```

Comment:
I think `outputDirectory` should be a required argument, don't think there is much use case for unzipping into an unknown directory? Or what if the full output path was returned?